apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,104 @@
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => QubitSetting::getByName('diacritics')->getValue(['sourceCulture' => true]),
+        ];
+    }
+
+    public function execute($request)
+    {
+        DefaultEditAction::execute($request);
+
+        if ($request->isMethod('post')) {
+            $this->form->bind($request->getPostParameters(), $request->getFiles());
+
+            if ($this->form->isValid()) {
+                if (0 === intval($request->getPostParameters()['diacritics']) || (1 === intval($request->getPostParameters()['diacritics']) && 0 !== $request->getFiles()['mappings']['size'])) {
+                    $this->processForm($request);
+                    $this->uploadDiacritics($request);
+
+                    $this->getUser()->setFlash('notice', $this->context->i18n->__('Diacritics setting updated. Rebuild the search index for the changes to be applied.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                } else {
+                    $this->getUser()->setFlash('error', $this->context->i18n->__('To enable diacritics setting, please upload a char_filter mapping YAML.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                }
+            }
+        }
+    }
+
+    public function processForm($request)
+    {
+        sfConfig::set('app_diacritics', $request->getPostParameters()['diacritics']);
+        QubitSetting::findAndSave('diacritics', $request->getPostParameters()['diacritics'], ['sourceCulture' => true]);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
+                // Move uploaded file to new location to pass off to background arFileImportJob.
+                try {
+                    $file = Qubit::moveUploadFile($file);
+                } catch (sfException $e) {
+                    $this->getUser()->setFlash('error', $e->getMessage());
+                }
+
+                // Replace /config/diacritics_mapping.yml with uploaded file
+                $diacriticsMappingPath = '/atom/src/plugins/arElasticSearchPlugin/config/diacritics_mapping.yml';
+                $uploadMappingPath = $file['tmp_name'];
+                $diacriticsMappings = file_get_contents($uploadMappingPath);
+                file_put_contents($diacriticsMappingPath, $diacriticsMappings);
+            }
+        } else {
+            // Reset diacritics yaml when disabling the setting
+            $diacriticsMappingPath = '/atom/src/plugins/arElasticSearchPlugin/config/diacritics_mapping.yml';
```

Comment:
This variable could maybe be set earlier in code as it ends up being set to the same thing twice.